### PR TITLE
feat: Add dsplit function across multiple backends

### PR DESCRIPTION
**Pull request by Shi-pra-19**
## Description

Implemented a new dsplit operation in the Keras ops API, enabling array splitting along the third axis (depth-wise splitting).The implementation ensures consistent behavior across all supported backends (TensorFlow, JAX, NumPy, OpenVINO, Torch).
## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -9811,3 +9811,53 @@ def unique(
...
+@keras_export(["keras.ops.dsplit", "keras.ops.numpy.dsplit"])
+def dsplit(x, indices_or_sections):
+    """Split an array into multiple sub-arrays depth-wise.
+
+    Args:
+        x: Input tensor.
+        indices_or_sections: If an integer, N, the tensor will be split into N
+            equal sections along axis 0. If a 1-D array of sorted integers,
+            the entries indicate indices at which the tensor will be split
+            along axis 0.
```
[the remark this answers is not in the document]
Comment (Shi-pra-19) [reply keras/src/ops/numpy.py]: Done.

**Review by hertschuh — APPROVED** [review]
Thanks for adding this!
